python/swe/examples/crewai_agent/agent.py

```diff
@@ -2,17 +2,72 @@
 
 # isort: skip_file
 
+from venv import create
 import dotenv
+import os
+import typing as t
 from composio_crewai import App, Action, ComposioToolSet, WorkspaceType
+from composio import action
 from crewai import Agent, Crew, Process, Task
-from langchain_openai import ChatOpenAI
+from langchain_openai import ChatOpenAI, AzureChatOpenAI
+from langchain_anthropic import ChatAnthropic
 from prompts import BACKSTORY, DESCRIPTION, EXPECTED_OUTPUT, GOAL, ROLE
 
 # Load environment variables from .env
 dotenv.load_dotenv()
 
+@action(toolname="math")
+def calculate_operation(num1: float, num2: float, operation: str) -> float:
+    """
+    Calculate the sum of two numbers
+    """
+    if operation == "add":
+        return num1 + num2
+    if operation == "subtract":
+        return num1 - num2
+    if operation == "multiply":
+        return num1 * num2
+    if operation == "divide":
+        return num1 / num2
+    raise ValueError(f"Invalid operation: {operation}")
+
 # Initialize tool.
-openai_client = ChatOpenAI(model="gpt-4o")  # type: ignore
+def get_langchain_llm() -> t.Union[ChatOpenAI, AzureChatOpenAI, ChatAnthropic]:
+    helicone_api_key = os.environ.get("HELICONE_API_KEY")
+    if os.environ.get("ANTHROPIC_API_KEY"):
+        if helicone_api_key:
+            print("Using Anthropic with Helicone")
+            return ChatAnthropic(
+                model_name="claude-3-5-sonnet-20240620",
+                anthropic_api_url="https://anthropic.helicone.ai/",
+                default_headers={
+                    "Helicone-Auth": f"Bearer {helicone_api_key}",
+                },
+            )  # type: ignore
+        print("Using Anthropic without Helicone")
+        return ChatAnthropic(model_name="claude-3-5-sonnet-20240620")  # type: ignore
+    if os.environ.get("OPENAI_API_KEY"):
+        if helicone_api_key:
+            print("Using OpenAI with Helicone")
+            return ChatOpenAI(
+                model="gpt-4-turbo",
+                base_url="https://oai.helicone.ai/v1",
+                default_headers={
+                    "Helicone-Auth": f"Bearer {helicone_api_key}",
+                },
+            )
+        print("Using OpenAI without Helicone")
+        return ChatOpenAI(model="gpt-4-turbo")
+    if os.environ.get("AZURE_OPENAI_API_KEY"):
+        print("Using Azure OpenAI")
+        return AzureChatOpenAI(model="test")
+
+    raise RuntimeError(
+        "Could not find API key for any supported LLM models, "
+        "please export either `ANTHROPIC_API_KEY`, `OPENAI_API_KEY` "
+        "or `AZURE_OPENAI_API_KEY`"
+    )
+
 composio_toolset = ComposioToolSet(workspace_config=WorkspaceType.Docker())
 
```

Comment:
The function `get_langchain_llm` in `agent.py` has multiple print statements which could clutter the output logs. Consider using a logging framework with appropriate log levels for better control over logging output.